### ISSUE TITLE
Request: Mvn Repository

### ISSUE THREAD

**Issue by mwong56** [binance-exchange/binance-java-api#3]
Can we get this added to maven repository?

**Comment by joaopsilva** [comment]
That was indeed in the plans, I will look into putting the library available from Maven Central in the next few days.

**Comment by mwong56** [comment]
Awesome! I've been using this for now https://jitpack.io/#binance-exchange/binance-java-api

**Comment by joaopsilva** [comment]
JitPack is a great alternative indeed. In thea meantime, I have created a ticket with Sonatype (https://issues.sonatype.org/browse/OSSRH-36611), and if they are able to host the binaries, I will then synchronise them to Maven Central.

**Comment by aparsons** [comment]
+1 to getting this done

**Comment by guitcastro** [comment]
+1

**Comment by klaszlo8207** [comment]
+1

**Comment by yiqinghe** [comment]
+1

**Comment by klaszlo8207** [comment]
The jitpack url is working like a charm. I am using with gradle and android with full of luck:)

**Comment by AslanKralCan** [comment]
+1

**Comment by vladimir-shulga** [comment]
+1

**Comment by datalorax** [comment]
+1

**Comment by zagorulkinde** [comment]
+1

**Comment by Zap-Dev** [comment]
+1

**Comment by mustii82** [comment]
Still not available please upload it and make a jar version to easy import this Library

**Comment by naeem-github** [comment]
Please do this ASAP

**Comment by vladimirus** [comment]
in the meantime https://jitpack.io/ is quite good
```xml
       <repositories>
		<repository>
		    <id>jitpack.io</id>
		    <url>https://jitpack.io</url>
		</repository>
	</repositories>
```
```xml
       <dependency>
	    <groupId>com.github.binance-exchange</groupId>
	    <artifactId>binance-java-api</artifactId>
	    <version>master-SNAPSHOT</version>
	</dependency>
```

**Comment by big-andy-coates** [comment]
Any update on this @joaopsilva?

**Comment by afaslan** [comment]
+1

**Comment by tach1685** [comment]
+1

**Comment by carvalho2707** [comment]
+1

**Comment by tadaskrisciunas** [comment]
+1

**Comment by yaserhadi** [comment]
+1

**Comment by noonhub** [comment]
The [README](https://github.com/binance-exchange/binance-java-api#installation) seems to claim this is in Maven but a search only shows other libraries and a fork. ¯\_(ツ)_/¯

**Comment by tach1685** [comment]
Hi joaopsilva, Any news on where this issue is on the roadmap?

**Comment by malkusch** [comment]
@noonhub No it doesn't:

> Install library into your Maven's local repository by running mvn install

**Comment by wrone** [comment]
+1

**Comment by adcar** [comment]
+1

**Comment by yqsailor** [comment]
+1

**Comment by antonvalletas** [comment]
+1

**Comment by mutellipiqbal** [comment]
Maven Pom file could not find dependency version. 
![version fualt](https://user-images.githubusercontent.com/32098494/53290378-1e423b80-3771-11e9-9b6e-9c517eca2595.png)

**Comment by guicastrofer** [comment]
+1 
I have the same problem on my pom.xml, mentioned above.

**Comment by joaovitormarques** [comment]
Hello @joaopsilva , any news on this?

**Comment by aparsons** [comment]
I ran into a problem where my dependency was cached and the API calls were broken. Adding `-SNAPSHOT` to the dependency version [will ensure you're using the latest version from jitpack](https://jitpack.io/docs/#snapshots).
```gradle
repositories {
  maven { url "https://jitpack.io" }
}

dependencies {
  implementation 'com.github.binance-exchange:binance-java-api:master-SNAPSHOT'
}
```

On a side note, @joaopsilva please fix this. [It's easy to do!](https://maven.apache.org/repository/guide-central-repository-upload.html)

**Comment by freekode** [comment]
3 years and still open?

**Comment by rotilho** [comment]
Any progress on it? It should be a simple task.

**Comment by chinnaxs** [comment]
Also in favor of getting this to the maven repo

**Comment by omerg** [comment]
I've forked the project and then published to Github Packages. You can now load it with Maven with the following dependency:

<dependency>
  <groupId>com.binance.api</groupId>
  <artifactId>binance-api-client</artifactId>
  <version>1.0.1</version>
</dependency>

Here is the link to package page:
https://github.com/LucidCodeTr/binance-java-api/packages/586895

**Comment by carlspring** [comment]
Any updates on making this available from Maven Central?

**Comment by adkr** [comment]
Shame... Almost 4 years and this issue is still opened.
That's really bad from technical, public relations and brand perspective.

**Comment by infinic** [comment]
+1

**Comment by kwladyka** [comment]
Any rationale why the library is not in mvn?

**Comment by carlspring** [comment]
@kwladyka ,

Yes, the rationale is perfectly clear -- this project has been dead for a long while.

**Comment by kwladyka** [comment]
The project lucks of "deprecated" word in readme.md or should be archived in github. It is not obvious in first steps with Binance API.

**Comment by carlspring** [comment]
@kwladyka : If you check the commit history, you'll see that there has been no real activity on the project since it won the bounty for a Java implementation of a library for Binance. I don't think that the original author cares about further developing this. You can easily arrive to the same conclusion, if you have a look at the number of pull requests which have been made by various contributors and have not been reviewed, or merged.

**Comment by kwladyka** [comment]
I know. Just I am saying it is not friendly approach for developers :)

**Comment by carlspring** [comment]
@kwladyka : I agree, :)

**Comment by joaopsilva** [comment]
Hi @kwladyka , @carlspring , I did try to have the project on Maven right from the start, but since the `groupId` was set to `com.binance.api`, it needed the approval from someone at Binance (i.e. an email sent from "@binance.com"), and even though I requested it, the email was never sent. So moving forward the `groupId` would need to be renamed, or we would need to press Binance again.

Another issue is given the sensitivity of this API, and how it can have full access to your account for trading purposes, it's usually recommended to run a local audited copy instead of just blindly trusting a binary from Maven, but ultimately that depends on the trust-level of the developer, and I agree there should be an option.

Finally, I have been busy with multiple other projects since I released this API, and I no longer have time to actively maintain it, so I would happily accept new developers to take over; in case you are interested, please let me know, I will open an issue soon about it.
